internal/fuse/snapshots_dirstruct.go

```diff
@@ -302,10 +303,20 @@ func (d *SnapshotsDirStructure) updateSnapshots(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	// sort snapshots ascending by time (default order is descending)
+
+	// Sort snapshots ascending by time (default order is descending).
+	// This needs to be done before hashing.
 	sort.Sort(sort.Reverse(snapshots))
 
-	if d.snCount == len(snapshots) {
+	// We update the snapshots when the hash of their id's changes.
+	h := sha256.New()
+	for _, sn := range snapshots {
+		h.Write(sn.ID()[:])
+	}
+	var hash [sha256.Size]byte
+	h.Sum(hash[:0])
+
+	if d.lastCheck != (time.Time{}) && d.hash == hash {
```

Comment:
Just checking whether the hash matches would probably be enough. Otherwise we'd have a hash collision.

Reply:
Yes, we'd need to have a set of snapshots with an all-0 SHA2. Removed the timestamp check.